### ISSUE TITLE
Vendre BB haute pour TP bb basses

### ISSUE THREAD

**Issue by AdrienVansuyt** [AdrienVansuyt/Quant-Algo#7]
> from AlgorithmImports import *
> 
> class BearishBollingerBandsStrategy(QCAlgorithm):
>     def Initialize(self):
>         self.SetStartDate(2020, 1, 1)    # Set Start Date
>         self.SetEndDate(2021, 1, 1)      # Set End Date
>         self.SetCash(100000)             # Set Strategy Cash
>         self.symbol = self.AddEquity("SPY", Resolution.Daily).Symbol
>         
>         # Configuration des EMA pour détecter la tendance baissière
>         self.short_ema = self.EMA(self.symbol, 10, Resolution.Daily)  # EMA rapide
>         self.long_ema = self.EMA(self.symbol, 40, Resolution.Daily)  # EMA lente
> 
>         # Configuration des bandes de Bollinger
>         self.bollinger_bands = self.BB(self.symbol, 20, 3, MovingAverageType.Simple, Resolution.Weekly)
> 
>     def OnData(self, data):
>         if not (self.short_ema.IsReady and self.long_ema.IsReady and self.bollinger_bands.IsReady):
>             return
> 
>         price = data[self.symbol].Price
>         upper_band = self.bollinger_bands.UpperBand.Current.Value
>         lower_band = self.bollinger_bands.LowerBand.Current.Value
>         
>         # Vérifier la tendance baissière
>         if self.short_ema.Current.Value < self.long_ema.Current.Value:
>             # Entrée en position si le prix touche ou casse la bande supérieure
>             if price >= upper_band and not self.Portfolio[self.symbol].IsShort:
>                 self.SetHoldings(self.symbol, -1)  # Entrer en position courte
>                 self.Debug(f"Going short at {price}")
>                 
>                 # Définir un take profit et un stop loss
>                 self.take_profit = lower_band
>                 self.stop_loss = price + 2.00  # 200 pips stop loss
>                 
>         elif self.Portfolio[self.symbol].Invested:
>             # Vérifier pour prendre des profits ou stopper les pertes
>             if price <= self.take_profit:
>                 self.Liquidate(self.symbol)
>                 self.Debug(f"Taking profit at {price}")
>             elif price >= self.stop_loss:
>                 self.Liquidate(self.symbol)
>                 self.Debug(f"Stopping loss at {price}")from AlgorithmImports import *
> 
> class BearishBollingerBandsStrategy(QCAlgorithm):
>     def Initialize(self):
>         self.SetStartDate(2020, 1, 1)    # Set Start Date
>         self.SetEndDate(2021, 1, 1)      # Set End Date
>         self.SetCash(100000)             # Set Strategy Cash
>         self.symbol = self.AddEquity("SPY", Resolution.Daily).Symbol
>         
>         # Configuration des EMA pour détecter la tendance baissière
>         self.short_ema = self.EMA(self.symbol, 10, Resolution.Daily)  # EMA rapide
>         self.long_ema = self.EMA(self.symbol, 40, Resolution.Daily)  # EMA lente
> 
>         # Configuration des bandes de Bollinger
>         self.bollinger_bands = self.BB(self.symbol, 20, 3, MovingAverageType.Simple, Resolution.Weekly)
> 
>     def OnData(self, data):
>         if not (self.short_ema.IsReady and self.long_ema.IsReady and self.bollinger_bands.IsReady):
>             return
> 
>         price = data[self.symbol].Price
>         upper_band = self.bollinger_bands.UpperBand.Current.Value
>         lower_band = self.bollinger_bands.LowerBand.Current.Value
>         
>         # Vérifier la tendance baissière
>         if self.short_ema.Current.Value < self.long_ema.Current.Value:
>             # Entrée en position si le prix touche ou casse la bande supérieure
>             if price >= upper_band and not self.Portfolio[self.symbol].IsShort:
>                 self.SetHoldings(self.symbol, -1)  # Entrer en position courte
>                 self.Debug(f"Going short at {price}")
>                 
>                 # Définir un take profit et un stop loss
>                 self.take_profit = lower_band
>                 self.stop_loss = price + 2.00  # 200 pips stop loss
>                 
>         elif self.Portfolio[self.symbol].Invested:
>             # Vérifier pour prendre des profits ou stopper les pertes
>             if price <= self.take_profit:
>                 self.Liquidate(self.symbol)
>                 self.Debug(f"Taking profit at {price}")
>             elif price >= self.stop_loss:
>                 self.Liquidate(self.symbol)
>                 self.Debug(f"Stopping loss at {price}")
